extensions/jackson/src/main/java/io/jsonwebtoken/jackson/io/JacksonDeserializer.java

```diff
@@ -35,6 +42,32 @@ public JacksonDeserializer() {
         this(JacksonSerializer.DEFAULT_OBJECT_MAPPER);
     }
 
+    /**
+     * Creates a new JacksonDeserializer where the values of the claims can be parsed into given types. A common usage
```

Comment:
Can we add some documentation here that it uses a new `ObjectMapper` instance?

For example, after the current documentation:

```
<p>
Because custom type parsing requires modifying Jackson {@code ObjectMapper} state, this 
constructor creates a new internal {@code ObjectMapper} instance and customizes it to support the 
specified {@code claimTypeMap}.  This ensures that JJWT parsing behavior does not unexpectedly 
modify the state of another application-specific {@code ObjectMapper}.
</p>
<p>
If you would like to use your own {@code ObjectMapper} instance that also supports custom types for 
JWT {@code Claims}, you will need to first customize your {@code ObjectMapper} instance by registering 
your custom types and then use the {@link JacksonDeserializer(ObjectMapper) } constructor instead.
</p>
```

Based on this, I wonder if we might want to make `MappedTypeDeserializer` a dedicated independent public class in the `jjwt-jackson` module so that users can use it on their own ObjectMapper instance (e.g. they might have customized pretty printing, but still want custom type registration, etc).  That class would document how to use it in the class JavaDoc based on what we've done, e.g.:

```java
MappedTypeDeserializer typeDeserializer = new MappedTypeDeserializer(typeMap);
SimpleModule module = new SimpleModule();
module.addDeserializer(Object.class, typeDeserializer);
objectMapper.registerModule(module);
```

All of this would help address the confusion I had (for example) around type customization on an application-wide deserializer, as well as provide an out-of-the-box solution for people that wanted to customize their own ObjectMapper.

Thoughts?

Reply:
Great doc suggestion

I thought about extracting `MappedTypeDeserializer` out, but I worry it might cause a maintenance headache trying to cover more parsing edge cases (when combined with other Jackson configurations).  My personal pref would be to hold off unless people ask for it (YAGNI?).  It's really easy to extract that class later, but if we make invalid assumptions now, it is harder to retract them.

That was my thinking anyway, though I could be missing some of the benefits of sharing an ObjectMapper instance?

Reply:
> I worry it might cause a maintenance headache trying to cover more parsing edge cases (when combined with other Jackson configurations)

I was thinking the current implementation would be the only one we'd support, and if that doesn't meet one's needs, they can just do it the manual approach or implement their own `JsonSerializer`.  At least they'd see easily how to do it themselves since understanding how that works is most of the difficulty, and our example gets them out of the weeds.

That said, I see value with leaving it private and the YAGNI argument.  I guess I'm on the fence 50/50 and would be happy with whatever you choose, so feel free ;)